omod-2.2/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_2/ConditionController2_2Test.java

```diff
@@ -161,6 +163,77 @@ public void shouldCreateACodedCondition() throws Exception {
 		Assert.assertNotNull(newConditionSource.get("onsetDate"));
 	}
 	
+	@Test
+	public void shouldUpdateANonCodedCondition() throws Exception {
+		String conditionUuid = RestTestConstants2_2.NON_CODED_CONDITION_UUID;
+		Condition condition = conditionService.getConditionByUuid(conditionUuid);
+		
+		Assert.assertNotNull(condition);
+		
+		String oldCondition = condition.getCondition().getNonCoded();
+		
+		SimpleObject codedOrFreeText = new SimpleObject();
+		codedOrFreeText.add("nonCoded", "Updated non coded condition");
+		
+		SimpleObject attributes = new SimpleObject();
+		attributes.add("condition", codedOrFreeText);
+		attributes.add("clinicalStatus", ConditionClinicalStatus.ACTIVE);
+		attributes.add("verificationStatus", ConditionVerificationStatus.CONFIRMED);
+		attributes.add("onsetDate", "2018-02-23 00:00:00");
+		attributes.add("voided", false);
+		
+		String json = new ObjectMapper().writeValueAsString(attributes);
+		
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI() + "/" + conditionUuid);
+		req.setContent(json.getBytes());
+		handle(req);
+		
+		condition = conditionService.getConditionByUuid(conditionUuid);
+		
+		Assert.assertNotNull(condition);
+		Assert.assertNotEquals(oldCondition, condition.getCondition().getNonCoded());
+		Assert.assertEquals(ConditionClinicalStatus.ACTIVE, condition.getClinicalStatus());
+		Assert.assertEquals(ConditionVerificationStatus.CONFIRMED, condition.getVerificationStatus());
+		Assert.assertNotNull(condition.getOnsetDate());
+	}
+	
+	@Test
+	public void shouldUpdateACodedCondition() throws Exception {
+		String conditionUuid = RestTestConstants2_2.CODED_CONDITION_UUID;
```

Comment:
Why do you assign it to a variable if it is already a constant?

Reply:
😅 I had completely overlooked that. Let me correct it.